syntax = "proto3";

package common;

// Common timestamp message for all services
message Timestamp {
  int64 seconds = 1;
  int32 nanos = 2;
}

// User context for authentication/authorization
message UserContext {
  string user_id = 1;
  repeated string roles = 2;
}

// Empty response message
message Empty {}

// Standard status response
message Status {
  bool success = 1;
  string message = 2;
  int32 code = 3;
}
